### PR TITLE
[UNR-5108] Fixed regex used to search for IDs during schema generator tests

### PR DESCRIPTION
**Pull request by martin-improbable**
#### Description
The regex used to search for IDs (during said tests) enforced tabs, which are optional in schema files. The tabs have therefore been removed from the regex, to avoid missing IDs during parsing.

#### Tests
Ran automated tests.

### CODE REVIEW

**Review by simonsarginson — APPROVED** [review]
Straightforward LGTM

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ ComponentNamesAndIds ParseAvailableNamesAndIdsFromSchemaFile(const TArray<FStrin
 
 	for (const auto& SchemaLine : LoadedSchema)
 	{
-		FRegexPattern IdPattern(TEXT("(\tid = )([0-9]+)(;)"));
+		FRegexPattern IdPattern(TEXT("(id = )([0-9]+)(;)"));
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:63]: Are strict on white spaces in the rest of the statement? E.g should it be:
`"(id/s*=/s*)(/s*[0-9]+)(/s*;)"`

Comment (simonsarginson) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:63]: Also not entirely sure why have three capture groups but this is turning into a digression that is probably out of scope.

Comment (martin-improbable) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:63]: Yes, especially the trailing semi-colon capture group - but I won't tackle these issues in this PR

**Review by improbable-danny — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp [hunk 2]
```diff
@@ -60,7 +60,7 @@ ComponentNamesAndIds ParseAvailableNamesAndIdsFromSchemaFile(const TArray<FStrin
 
 	for (const auto& SchemaLine : LoadedSchema)
 	{
-		FRegexPattern IdPattern(TEXT("(\tid = )([0-9]+)(;)"));
+		FRegexPattern IdPattern(TEXT("([\t\\s]id\\s*=\\s*)([0-9]+)(\\s*;)"));
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:63]: `[\t\\s]` is this strictly required?

Comment (simonsarginson) [reply SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp:63]: Ah fair. You don't need \t \\s should match it.

**Pull request merged**